data/dishHelpers.js

```diff
@@ -0,0 +1,57 @@
+const knex = require("knex");
+const dbConfig = require("../knexfile.js");
+const db = knex(dbConfig.development);
+
+module.exports = {
+	getDishes: () => {
+		return db("dishes");
+	},
+	addDish: dish => {
+		return db("dishes")
+			.insert(dish)
+			.then(ids => ids[0]);
+	},
+	getDish: id => {
+		return db
+			.select("dish_name", "recipe_name")
+			.from("dishes")
+			.where({ "dishes.id": id })
+			.innerJoin("recipes", "recipes.dish_id", "dishes.id");
```

Comment:
There can be multiple recipes for a single dish, so does this represent that? If there are multiple columns named the same thing, the column will just get overwritten (so only one value will be there). Really, what you'd want to do here is do another query for recipes based on their dish ids and stick them in an array (with 'recipes' as the key probably) in the dish object, like this:
```javascript
// a dish
{
  // ...
  dish_name: "tacos",
  recipes: [
    // Either recipe objects, 
    // or just their names as strings
    { // a recipe },
    { // another recipe },
    // ...
  ]
}
```

Reply:
This makes perfect sense. Thank you. I ended up confronting this in the sprint challenge today. However, I did this with with javascript in the routes sections. Is there a better way to do this with SQL/Knex?

Reply:
You theoretically could do this in SQL, but it's a lot more readable in JavaScript. Now, in terms of performance, usually there's a benefit for just using Knex/SQL since you can reduce it to one query, but I personally say that the most likely slight performance hit of using two queries is definitely worth it for the sake of readability. As for the SQL implementation, I'm not entirely sure what it'd take (but, as said, I'm fairly sure you _could_ do it)!